liota/core/package_manager.py

```diff
@@ -30,4 +30,1104 @@
+                        return None
+                    if not dependency in self._packages_loaded:
+                        self._package_load(dependency, check_stack=check_stack)
+                    if not dependency in self._packages_loaded:
+                        log.error("%s is not loaded, because %s failed to load"\
+                                % (file_name, dependency))
+                        check_stack.pop()
+                        return None
+                    
+                    # Add dependent record
+                    dep_record = self._packages_loaded[dependency]
+                    assert(isinstance(dep_record, PackageRecord))
+                    dep_record.add_dependent(file_name)
+                check_stack.pop()
+                log.debug("Dependency check of package %s is complete" \
+                        % file_name)
+
+        # Get package class from module and instantiate it
+        if not hasattr(module_loaded, "PackageClass"):
+            log.error("Invalid package: %s" % module_name)
+            return None
+
+        klass = getattr(module_loaded, "PackageClass")
+        package_record = PackageRecord(file_name)
+        if not package_record.set_instance(klass()):
+            log.error("Unexpected failure initializing package")
+            return None
+        try: # Run created instance
+            package_record.get_instance().run(
+                    self._resource_registry.get_package_registry(file_name)
+                )
+        except Exception as er:
+            log.error("Exception in initialization: %s" % str(er))
+            return None
+        package_record.set_sha1(sha1)
+        package_record.set_ext(file_ext)
+        package_record.set_dependencies(dependencies)
+        self._packages_loaded[file_name] = package_record
+
+        log.info("Package class from module %s is initialized" \
+                % module_loaded.__name__)
+        return package_record
+
+    #-----------------------------------------------------------------------
+    # This method is called to unload package using its file_name (no ext).
+
+    def _package_unload(self, file_name, track_list=None):
+        log.debug("Attempting to unload package: %s" % file_name)
+        
+        # Check if specified package is already loaded
+        if not file_name in self._packages_loaded:
+            log.warning("Could not unload package - not loaded: %s" \
+                    % file_name)
+            return False
+
+        package_record = self._packages_loaded[file_name]
+        assert(isinstance(package_record, PackageRecord))
+
+        # Stop all dependents, before making any change to current package
+        dependents = package_record.get_dependents()
+        
+        if len(dependents) > 0:
+            log.info("Package %s is depended by: %s" \
+                    % (file_name, " ".join(dependents)))
+            for dependent in dependents:
```

Comment:
just need to mention package 2 (when load package 2, package manager will check package 1).
From package examples, it's easy to realize it.